egs2/how2/asr1/local/data.sh

```diff
@@ -0,0 +1,207 @@
+    text=${3}
+    pattern=${4}
+    new_path=${5}
+
+    [ ! -d ${srcdir} ] && log "$0: No such directory ${srcdir}" && exit 2
+    [ -d ${dstdir} ] && rm -rf ${2}
+    mkdir -p ${workdir}
+
+    check_files ${srcdir} ${text} && cp -r ${srcdir}/* ${dstdir}
+
+    # re-write paths in feats.scp and wav.scp with user paths
+    for x in feats.scp cmvn.scp; do
+        if [[ ${srcdir} == /* ]]; then
+            sed -i "s|${pattern}|${new_path}|g" ${dstdir}/${x}
+        else
+            sed -i "s|${pattern}|${PWD}/${new_path}|g" ${dstdir}/${x}
+        fi
+    done
+
+    cut -d' ' -f1 ${dstdir}/${text} > ${workdir}/id.en
+    cut -d' ' -f2- ${dstdir}/${text} > ${workdir}/text.en
+
+    # normalize punctuation using moses decoder
+    normalize-punctuation.perl -l en < ${workdir}/text.en > ${workdir}/text.en.norm
+
+    # normalize text and modify other needed files (utt2spk, spk2utt..) according to new text file
+    # excluding punctiation normalization iwslt19 test set remains unchanged.
+    if [[ ${dstdir} == *"test_set_iwslt2019"* ]]; then
+        cp ${workdir}/id.en ${workdir}/id.en.norm
+        cp ${workdir}/text.en.norm ${workdir}/text.en.norm2
+    else
+        local/normalize_how2_texts.sh ${workdir}/text.en.norm ${workdir}/id.en \
+                                      ${workdir}/text.en.norm2 ${workdir}/id.en.norm
+
+        # utterances removed during text normalization are also removed in other files
+        for x in segments utt2spk feats.scp; do
+            match_files ${workdir} ${x} ${dstdir}
+        done
+    fi
+
+    # saving lowercase and uppercase version of text. uppercase is for iwslt19
+    tr '[:upper:]' '[:lower:]' < ${workdir}/text.en.norm2 > ${workdir}/text.en.norm2.lc
+    tr '[:lower:]' '[:upper:]' < ${workdir}/text.en.norm2 > ${workdir}/text.en.norm2.uc
+
+    paste -d' ' ${workdir}/id.en.norm ${workdir}/text.en.norm2.lc > ${dstdir}/text
+    paste -d' ' ${workdir}/id.en.norm ${workdir}/text.en.norm2.uc > ${dstdir}/text.uc
+
+    check_files ${dstdir} text
+}
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    log "stage 0: Data preparation (HOW2 + IWSLT 2019 test set)"
+
+    feats_path=${data_how2_text}/features
+    feats_pattern="ARK_PATH"
+
+    for set in train val dev5 test_set_iwslt2019; do
+        srcdir=${data_how2_text}/data/${set}
+        dstdir=data/${set}
+        text="text.id.en"
+
+        if [[ $set == "test_set_iwslt2019" ]]; then
+            feats_path=${data_iwslt19}
+            feats_pattern="\.\."
+            srcdir=${data_iwslt19}/test
+            text="text"
+        fi
+
+        prepare_set ${srcdir} ${dstdir} ${text} ${feats_pattern} ${feats_path}
+
+        if [ $set == "train" ]; then
+            remove_longshortdata.sh --maxframes ${maxframes}        \
+                                    --maxchars ${maxchars}          \
+                                    data/${set}                     \
+                                    data/${set}_reduced
+        fi
```

Comment:
I'll add `remove_longshortdata.sh` in asr.sh . We discussed in #1504, but it's o.k. for this recipe. Just a note.